apps/meteor/client/components/Omnichannel/CustomFieldsForm/index.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { ReactElement } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { CustomSelectInput } from './CustomSelectInput';
+import { CustomTextInput } from './CustomTextInput';
+import { useCustomFields } from './hooks/useCustomFields';
+
+const CustomFieldsForm = (): ReactElement | null => {
+	const result = useCustomFields();
+
+	const { register, control } = useFormContext<{ livechatData: Record<string, unknown> }>();
+
+	const { data, isLoading } = result;
+
+	if (!data || isLoading) {
+		return null;
+	}
+
+	if (result.isError) {
+		throw result.error;
+	}
+
+	return (
+		<>
+			{data.customFields.map((customField) => {
+				// TODO: REMOVE FILTER ONCE THE ENDPOINT SUPPORTS A SCOPE PARAMETER
+				if (customField.scope !== 'room') {
+					return null;
+				}
+
+				switch (customField.type) {
+					case 'input':
+						return (
+							<CustomTextInput
+								key={customField._id}
+								data={customField}
+								{...register(`livechatData.${customField._id}`, {
+									validate: {
+										required: (value) => !!value,
+										regexp: (value) => !!String(value).match(new RegExp(customField.regexp)),
+									},
+								})}
+							/>
+						);
+					case 'select':
+						return (
+							<Controller
+								key={customField._id}
+								control={control}
+								name={`livechatData.${customField._id}`}
+								render={({ field }): ReactElement => <CustomSelectInput data={customField} {...field} />}
+								rules={{
+									validate: {
+										required: (value) => !!value,
+									},
+								}}
+							/>
```

Comment:
Why does `CustomSelectInput` need a controller?

Reply:
because I can't control custom inputs like our Select Component without it...